packages/ui-spinner/src/Spinner/index.tsx

```diff
@@ -148,6 +164,10 @@ class Spinner extends Component<SpinnerProps> {
       </View>
     )
   }
+
+  render() {
+    return this.state.shouldRender ? this.renderSpinner() : null
```

Comment:
I would still render a transparent rectangle when`shouldRender` is false to prevent "pop in" and potential realignment of the UI. Maybe render the svg with `visibility: hidden`?

Reply:
This would defeat the purpose of the delay, since the goal of the feature is to not disrupt the UI if the load time is small